packages/datatrak-web/src/features/Survey/utils/utils.ts

```diff
@@ -67,3 +68,40 @@ export const getErrorsByScreen = (
     }, {}) ?? {}
   );
 };
+
+const hasEntityQuestionConfig = (
+  ssc: SurveyScreenComponent,
+): ssc is SurveyScreenComponent & {
+  config: { entity: EntityQuestionConfig };
+} =>
+  (ssc.type === QuestionType.Entity || ssc.type === QuestionType.PrimaryEntity) &&
+  ssc.config?.entity !== undefined;
+
+export const getParentQuestionId = (question: SurveyScreenComponent) =>
+  hasEntityQuestionConfig(question) && question.config.entity?.filter?.parentId?.questionId;
+
+export const getPrimaryQuestionAncestorAnswers = (question, questionsById, ancestorsByType) => {
+  const filterType = question?.config?.entity?.filter?.type[0];
+  const ancestor = ancestorsByType[filterType];
+  const record = {
+    [question.id]: ancestor.id,
+  };
+  const parentQuestionId = getParentQuestionId(question);
+  if (!parentQuestionId) {
+    return record;
+  }
+  const parentQuestion = questionsById[parentQuestionId];
```

Comment:
Potentially need to also handle the off chance something is configured wrong and the parent question can't be found

Reply:
Thanks for picking all these up. I did a bit of refactoring to better handle this parsing including: adding Types, checking for false values and added a test. Would you be able to just do a quick re-review of that?